### ISSUE TITLE
Chat

### ISSUE THREAD

**Issue by SeniorCluckers** [SeniorCluckers/NCL2#1]
@essem2 

Sorry don't know where to msg you. But you said we can save the texture. Humm okay so bottom line is..
if you wanted it to be possible.

1. You need accounts where the skin is the link the user wants and then it is uploaded to those accounts.
2. Then we take the skin from that account and save the skin texture and signature to the server.
3. Then we set it?

Idk just taking a guess.

**Comment by essem2** [comment]
yes, i dont know if i understand 1. correctly but you need just one account because you can change the skin after you saved the string.

**Comment by essem2** [comment]
here is an example how i do it on a plugin for my server (for security & privacy reasons i removed some parts) http://i.imgur.com/fPDl9XI.png

**Comment by SeniorCluckers** [comment]
Wait you do it using bungeecord? You change the players skin on login event? To a custom skin of their choosing? Or too a particular skin?

**Comment by essem2** [comment]
I use bungeecord but not the skin-api from bungeeutil because idk. i change the players skin on preloginevent for one hardcoded player with this hardcoded skin.

**Comment by SeniorCluckers** [comment]
Would you say you feel better about it rather than using the API?

**Comment by essem2** [comment]
For someone without experience in reflection and stuff like this its safer to use the API. i wrote this stuff before the api existed.

**Comment by SeniorCluckers** [comment]
Ah okay. So I tested the custom skin plugin. And it does indeed save the info to your server files, therefore 🤔  can't I use the skin data that I got from the plugin ? Obviously players still would not be able to use their own links but I would be able to set pre-defined skins right? 

`{"id":"10ceea5b-5c44-350b-97b0-1fed6817f3ac","name":"","properties":[{"name":"textures","value":"eyJ0aW1lc3RhbXAiOjE0OTk4MzY0MzY5ODksInByb2ZpbGVJZCI6IjQzYTgzNzNkNjQyOTQ1MTBhOWFhYjMwZjViM2NlYmIzIiwicHJvZmlsZU5hbWUiOiJTa3VsbENsaWVudFNraW42Iiwic2lnbmF0dXJlUmVxdWlyZWQiOnRydWUsInRleHR1cmVzIjp7IlNLSU4iOnsidXJsIjoiaHR0cDovL3RleHR1cmVzLm1pbmVjcmFmdC5uZXQvdGV4dHVyZS9iNWE1M2E0NGZkN2JjNGI0ZDdiYTI2Yzc0ODQ4YjVlNDQzODkyNWNmM2Q4YzdiNGI4ZDQzMTFlOGI2ZmEifX19","signature":"ouCK7KLayEL4lEdq9NfhgubICzbl6CqOYvBeUmVpeCZX+P9TKg7Lc7EgY44+FKHUxjupS2Hrof9BvNORfBUWnLkg27ymasp+6AYy1K5S6K/l1o859ztgjirGHQBHdG66r2MoGtxUX+y67IY1mQFgfH++31JcBlOkpRE9PWmViJySSzgFSc33mgp+K26jlHKbT7emh/1t/vrJsANnz/cv3rhL84npoT33hjP/ibjMUY7cyWKkAuFOIMNc1c35McBGzoCoXxA/WsK9gUhi9143L0cn4Zh8Od7t+n48uwww+8d28/3J3uwiX+fQBtfWj0QBtfyNbAC1Rqe1fMJt6TeNakTrn9/q+phTEmsDCr5pkbNFPB2Teu+41aqo2BV2LVoht+BFQQ/ojP/w+6PQ4W0Axe+tBal/JZ3ncPLmyVDzSWDmYoG0tW0FLWSmCMPsoqVjmoYisk/vzEMt7zhdDX1Ji+yKxPJe5b8YRKvbZkDF8YN264G2HulUzEMARe/nGSiIzIUTSAWiGaJlr0r6DGvVISMZephCNUek1DyFJzMuSdfzAW7G7ocTPEeh4+bLflFWf8iYNjy2GarlSNJxojflDAvQ5T9mTWhFzHi8X139D6a9bUuMUatDFplUqohgBn3XNAfIJ+/qZRhk/I2LWEw9efCojmiJEQmA727mjyidToo\u003d"}]}`

**Comment by essem2** [comment]
yes you would, its the same what i do but just for one player. by the way you can access the data you need via your own plugin or via https://sessionserver.mojang.com/session/minecraft/profile/<uuid>

**Comment by SeniorCluckers** [comment]
Not sure what you mean exactly but this is what I tried and failed.

`	public void execute(CommandSender sender, String[] args) {

		ProxiedPlayer player = ProxyServer.getInstance().getPlayer(sender.getName());
		
		String RawData = "eyJ0aW1lc3RhbXAiOjE0OTk4MzY0MzY5ODksInByb2ZpbGVJZCI6IjQzYTgzNzNkNjQyOTQ1MTBhOWFhYjMwZjViM2NlYmIzIiwicHJvZmlsZU5hbWUiOiJTa3VsbENsaWVudFNraW42Iiwic2lnbmF0dXJlUmVxdWlyZWQiOnRydWUsInRleHR1cmVzIjp7IlNLSU4iOnsidXJsIjoiaHR0cDovL3RleHR1cmVzLm1pbmVjcmFmdC5uZXQvdGV4dHVyZS9iNWE1M2E0NGZkN2JjNGI0ZDdiYTI2Yzc0ODQ4YjVlNDQzODkyNWNmM2Q4YzdiNGI4ZDQzMTFlOGI2ZmEifX19";
		String SignatureData = "ouCK7KLayEL4lEdq9NfhgubICzbl6CqOYvBeUmVpeCZX+P9TKg7Lc7EgY44+FKHUxjupS2Hrof9BvNORfBUWnLkg27ymasp+6AYy1K5S6K/l1o859ztgjirGHQBHdG66r2MoGtxUX+y67IY1mQFgfH++31JcBlOkpRE9PWmViJySSzgFSc33mgp+K26jlHKbT7emh/1t/vrJsANnz/cv3rhL84npoT33hjP/ibjMUY7cyWKkAuFOIMNc1c35McBGzoCoXxA/WsK9gUhi9143L0cn4Zh8Od7t+n48uwww+8d28/3J3uwiX+fQBtfWj0QBtfyNbAC1Rqe1fMJt6TeNakTrn9/q+phTEmsDCr5pkbNFPB2Teu+41aqo2BV2LVoht+BFQQ/ojP/w+6PQ4W0Axe+tBal/JZ3ncPLmyVDzSWDmYoG0tW0FLWSmCMPsoqVjmoYisk/vzEMt7zhdDX1Ji+yKxPJe5b8YRKvbZkDF8YN264G2HulUzEMARe/nGSiIzIUTSAWiGaJlr0r6DGvVISMZephCNUek1DyFJzMuSdfzAW7G7ocTPEeh4+bLflFWf8iYNjy2GarlSNJxojflDAvQ5T9mTWhFzHi8X139D6a9bUuMUatDFplUqohgBn3XNAfIJ+/qZRhk/I2LWEw9efCojmiJEQmA727mjyidToo\u003d";
		
		SkinFactory.createSkin(RawData, SignatureData);
		changePlayerSkin(player, RawData, SignatureData);

		sender.sendMessage(new TextComponent(ChatColor.GREEN + "You have changed your skin to Goofy's skin!" + args[0]));

	}

	void changePlayerSkin(ProxiedPlayer pplayer, String value, String signature) {
		LoginResult loginProfile = ((InitialHandler) pplayer.getPendingConnection()).getLoginProfile();
		loginProfile
				.setProperties(new LoginResult.Property[] { new LoginResult.Property("textures", value, signature) });

	}`

**Comment by essem2** [comment]
As of now it never happend.
Also if you do getInstance().getPlayer(sender) you do it overcomplicated and resource-demanding. you can just do 
if(sender instanceof ProxiedPlayer) {
ProxiedPlayer player = (ProxiedPlayer) sender;
Player p = (Player) player; //bungeeutils player if you need it
//do your stuff
} else {
//its not a player, its console or something
}

**Comment by SeniorCluckers** [comment]
@essem2 

Thank you I changed it to (ProxiedPlayer) sender instead.

**Comment by essemX** [comment]
is this repository still needed for chat? if you have questions, i am aviable via pn on spigotmc https://www.spigotmc.org/members/essem.26111/
